revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+        except ValueError:
+            pass
+    
+    """
+    Sets up the basics of a JSON serializable dictionary.
+    This sets up:
+    1. cash_in and its various components (repayments, adjustments, donations)
+    2. cash_out and its various components (reinvestments, adjustments)
+    3. date_info and its various components (start_date, end_date, date_list)
+    4. cash_balances
+    5. project_names
+    6. clean_project_names
+    7. cash_in/out_adjustment_names, which is a list of all the names of adjustments created
+
+    :return:
+        Returns a properly set up dictionary.
+    """
+    def setUpFilteredDict(self):
+        filtered_dict = {'cash_in': {'repayments': {}, 'adjustments': {}, 'donations': {}}, 'cash_out': {'reinvestments': {}, 'adjustments': {}}}
+        
+        filtered_dict['date_info'] = {}
+        filtered_dict['cash_balances'] = {}
+        date_info = filtered_dict['date_info']
+        date_info['start_date'] = self.start_date.strftime('%Y-%m-%d')
+        date_info['end_date'] = self.end_date.strftime('%Y-%m-%d')
+        date_info['date_list'] = []
+        filtered_dict['project_names'] = []
+        filtered_dict['clean_project_names'] = []
+        clean_project_names = filtered_dict['clean_project_names']
+        project_names = filtered_dict['project_names']
+        for proj in self.project_filter:
+            if (self.hasPayments(proj)):
+                project_names.append(proj.title)
+                clean_project_names.append(self.clean_project_name(proj.title))
+
+        filtered_dict['cash_in_adjustment_names'] = list(set([adj.name for adj in AdminAdjustment.objects.adjustments(start_date=self.start_date, end_date=self.end_date, cash_type='cash_in')]))
+        filtered_dict['cash_out_adjustment_names'] = list(set([adj.name for adj in AdminAdjustment.objects.adjustments(start_date=self.start_date, end_date=self.end_date, cash_type='cash_out')]))
+
+        return filtered_dict
+    
+    """
+    Checks whether a project has any associated donations, reinvestments, or repayments.
+
+    :args:
+        project: The project to check.
+
+    :return:
+        A boolean of whether the project has associated payment objects.
+    """
+    def hasPayments(self, project):
+        donations = len(Payment.objects.donations(project=project, start_date=self.start_date, end_date=self.end_date))
+        reinvestments = len(AdminReinvestment.objects.reinvestments(project=project, start_date=self.start_date, end_date=self.end_date))
+        repayments = len(AdminRepayment.objects.repayments(project=project, start_date=self.start_date, end_date=self.end_date))
+
+        return (donations + reinvestments + repayments) != 0
+
+    """
+    A helper function that returns a date one month ahead of the date
+    passed in.
+
+    :return:
+        A Python datetime.datetime object with a time value one month ahead of date.
+    """
+    def moveOneMonthForward(self, date):
```

Comment:
This one should definitely be in a utility somewhere, not in the View. In fact, what about putting all this computation stuff in a utility file, like a `utils.AccountingAggregator` class or something like that? There is a lot of logic in the view right now which would probably be easier to understand if it was extracted